File: KevinRene/src/qls/ast/widget/Slider.java
package qls.ast.widget;

import qls.ast.Widget;
import qls.ast.visitor.QLSVisitor;

public class Slider extends Widget {
	ValueSet values;
	
	public Slider(ValueSet values) {
		this.values = values;
	}
	
	@Override
	public <T> T accept(QLSVisitor<T> visitor) {
		return visitor.visit(this);
	}

	@Override
	public String toString() {
		return "Slider";
	}
}

File: guybas/QLS/AST/Widget/slider.py
import QLS.AST.Widget.widget_interface as w
from QL.AST.Expressions.Types import *


class Slider(w.IWidget):
    def __init__(self, min_value, max_value, default=""):
        self.min = min_value
        self.max = max_value
        self.default = default
        self._properties = {self.widget_name(): ""}

    def string_presentation(self, level=0):
        s = "    " * level + "Slider "
        s += self.min + " " + self.max
        s += "\n"
        return s

    def get_compatible(self):
        raise NotImplementedError("Not implemented by sub class")

    def get_range(self):
        return [number_type.Number()]

    def get_settings(self):
        return self._properties

    def set_settings(self, dictionary):
        for x in dictionary:
            self._properties[x] = dictionary[x]

    def widget_name(self):
        return "slider"